### PR TITLE
Add setting for threshold for reconnect on stalled stream.

### LINKED ISSUE #637: Disconnect from TVHeadend server while playing piped mpd-stream with "demux read no data for at least 10 secs"
Hi,

it might be a little bit specific, but I'm trying to play a mpd-stream via TVHeadend in Kodi. TVHeadend receives the stream as a pipe from Streamlink.

Playing it from VLC through TVHeadend works perfectly fine. The Kodi addon also receives the stream and plays it but frequently quits and the plugin disconnects from the TVHeadend server.

Kodi.log then states:
`2023-12-01 14:34:37.190 T:12154    info <general>: CVideoPlayerAudio::Process - stream stalled
2023-12-01 14:34:40.144 T:11577 warning <general>: AddOnLog: pvr.hts: demux read no data for at least 10 secs; restarting connection`

TVHeadend log:
`2023-12-01 14:32:51.156 htsp: ::ffff:192.168.178.47 [ stream | Kodi Media Center ]: Disconnected
2023-12-01 14:32:51.156 subscription: 051B: "::ffff:192.168.178.47 [ stream | Kodi Media Center ]" unsubscribing from "Sky Sport Bundesliga HD", hostname="::ffff:192.168.178.47", username="stream", client="Kodi Media Center"
2023-12-01 14:32:51.159 htsp: Got connection from ::ffff:192.168.178.47
2023-12-01 14:32:51.159 htsp: ::ffff:192.168.178.47: Welcomed client software: Kodi Media Center (HTSPv35)
2023-12-01 14:32:51.160 htsp: ::ffff:192.168.178.47 [ Kodi Media Center ]: Identified as user 'stream'`

Since VLC works fine it doesn't seem to be a problem on the server side but rather on the Kodi plugin. Is it possible to raise the timeout time (10 secs). I can also compile by myself to test.

### PR DESCRIPTION
**Pull request by ksooo**
Fixes #637

### CODE REVIEW

File: src/tvheadend/HTSPDemuxer.cpp [hunk 1]
```diff
@@ -149,7 +149,8 @@ DEMUX_PACKET* HTSPDemuxer::Read()
   }
   Logger::Log(LogLevel::LEVEL_TRACE, "demux read nothing");
 
-  if (m_lastPkt > 0 && m_lastUse - m_lastPkt > 10 && !IsPaused())
+  if (m_lastPkt > 0 && m_lastUse - m_lastPkt > m_settings->GetStreamStalledThreshold() &&
+      !IsPaused())
   {
     Logger::Log(LogLevel::LEVEL_WARNING,
                 "demux read no data for at least 10 secs; restarting connection");
```
Comment (Jalle19) [inline src/tvheadend/HTSPDemuxer.cpp]: The log message may be wrong if the user changes the default

File: src/tvheadend/InstanceSettings.cpp [hunk 2]
```diff
@@ -188,6 +193,8 @@ ADDON_STATUS InstanceSettings::SetSetting(const std::string& key,
     return SetStringSetting(GetStreamingProfile(), value);
   else if (key == "streaming_http")
     return SetBoolSetting(GetStreamingHTTP(), value);
+  else if (key == "streaming_stalledtreshold")
```
Comment (Jalle19) [inline src/tvheadend/InstanceSettings.cpp]: To be consistent with the code the setting should probably be named stream_stalled_threshold

File: src/tvheadend/HTSPDemuxer.cpp (hunk 1 above)
Comment (ksooo) [reply src/tvheadend/HTSPDemuxer.cpp]: Oops, thanks.

File: src/tvheadend/InstanceSettings.cpp (hunk 2 above)
Comment (ksooo) [reply src/tvheadend/InstanceSettings.cpp]: Will change that.

**Comment by ksooo** [comment]
@Jalle19 suggested changes are in. Good to go now?

BTW: Nice to see you being around again.

**Review by Jalle19 — APPROVED** [review]
Looks good 👌

**Pull request merged**